tempesta_fw/http_parser.c

```diff
@@ -1495,7 +1477,69 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len)
 		}
 	}
 
-	RGEN_LWS();
+	/* In request we should pass empty headers:
+	 * RFC 7230 5.4:
+	 * ....
+	 * ....
+	 * If the authority component is missing or
+	 * undefined for the target URI, then a client MUST send a Host header
+	 * field with an empty field-value.
+	 */
+	__FSM_STATE(RGen_LWS) {
+		switch (c) {
+		case '\r':
+			TFW_DBG3("\\r symbol\n");
+			if (parser->_header_tag >= 0) {
```

Comment:
There's no need to check for this case every time we process a header field. We know exactly when we need this particular case. For that reason even the flag mentioned above (or a new member of the parser structure, for that matter) is not needed.

It makes sense to leave the original `RGen_LWS` where it was, and replace it with a set of macros like the original `RGen_LWS()` and a new `RGen_LWS_empty()` that would block or pass a header with an empty value string. Most of the code in these macros is exactly the same, and one block of code will need to be parametrized.